File: Problemas/1742B - Increasing/1742B_Increasing.cpp
// 1742B - Increasing 
// https://codeforces.com/problemset/problem/1742/B

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {

        // Input:
        // 2) N = Número de elementos en el arreglo
        int N;
        cin >> N;

        // Input:
        // 3) arreglo = Números del arreglo
        int arreglo[101];
        for(int indice_arreglo = 0; indice_arreglo < N; indice_arreglo++) {
            cin >> arreglo[indice_arreglo];
        }

        // Set de números únicos
        set<int> numeros_unicos;

        // Itera sobre el arreglo, inserta los números en el set
        for(int indice_arreglo = 0; indice_arreglo < N; indice_arreglo++) {
            numeros_unicos.insert(arreglo[indice_arreglo]);
        }

        // Si el número de elementos en el arreglo es igual a la cantidad de números únicos en el mismo, entonces el arreglo se puede ordenar de manera ascendente estrictamente.
        // Imprime "YES" si las cantidades son iguales, de lo contrario, imprime "NO"
        if(numeros_unicos.size() == N) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}

File: Problemas/1742B - Increasing/1742B_Increasing.py
# 1742B - Increasing
# https://codeforces.com/problemset/problem/1742/B

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Input:
    # 2) N = Número de elementos en el arreglo
    N = int(input())

    # Input:
    # 3) arreglo = Números del arreglo
    arreglo = list(map(int, input().split()))

    # Recupera los números del arreglo
    numeros_unicos = set(arreglo)
    
    # Si el número de elementos en el arreglo es igual a la cantidad de números únicos en el mismo, entonces el arreglo se puede ordenar de manera ascendente estrictamente.
    # Imprime "YES" si las cantidades son iguales, de lo contrario, imprime "NO"
    if len(numeros_unicos) == len(arreglo):
        print("YES")
    else:
        print("NO")